File: samples/speech-recognition/from-file/python/transcription/transcript_files_by_restful_api.py
#!/usr/bin/env python3
# Copyright (c) alibaba.. All Rights Reserved.
#  MIT License  (https://opensource.org/licenses/MIT)
import json
import os
from http import HTTPStatus

import dashscope

# Set your DashScope API key. More information: https://help.aliyun.com/document_detail/2712195.html
if 'DASHSCOPE_API_KEY' in os.environ:
    dashscope.api_key = os.environ['DASHSCOPE_API_KEY']
    # in fact,if you have set DASHSCOPE_API_KEY in your environment variable,
    # you can ignore this, and the sdk will automatically get the api_key from the environment variable
else:
    dashscope.api_key = '<your-dashscope-api-key>'
    # if you can not set api_key in your environment variable,
    # you can set it here by code

# Submit the transcription task files list
# the transcription api supports most of the common audio formats
# you can check supported formats and other parameters here: https://help.aliyun.com/document_detail/2712535.html
# transcription api supports 100 files at most in one job, and each file size should be less than 2GB

# Submit the transcription task
task_response = dashscope.audio.asr.Transcription.async_call(
    model='sensevoice-v1',
    # 'paraformer-v1'
    file_urls=[
        'https://dashscope.oss-cn-beijing.aliyuncs.com/samples/audio/paraformer/hello_world_female2.wav',
        'https://dashscope.oss-cn-beijing.aliyuncs.com/samples/audio/paraformer/hello_world_male2.wav',
        'https://dashscope.oss-cn-beijing.aliyuncs.com/samples/audio/sensevoice/rich_text_example_1.wav',
        'https://dashscope.oss-cn-beijing.aliyuncs.com/samples/audio/sensevoice/rich_text_example_2.wav'
    ])
# This is the description of 'file_urls'.
# You need to provide a URL from which the file can be downloaded via HTTP.
# Typically, we can **store these files in public cloud storage services (such as Alibaba Cloud OSS)**
# and share a publicly accessible link.
# Note that it is best to add an expiration time to these links,
# to prevent third-party access if the file address is leaked.

# get the transcription result
transcribe_response = dashscope.audio.asr.Transcription.wait(
    task=task_response.output.task_id)
if transcribe_response.status_code == HTTPStatus.OK:
    print(json.dumps(transcribe_response.output, indent=4, ensure_ascii=False))
    print('transcription done!')
# you will get the transcription result in the transcribe_response.output by param : transcription_url
# transcription_url is a downloadable file of json format transcription result

File: samples/speech-recognition/recognize_speech_from_files_by_batch_mode/java/TranscriptFilesByRestfulApi.java
/*
 * Copyright (C) Alibaba Group. All Rights Reserved.
 * MIT License (https://opensource.org/licenses/MIT)
 */

package org.aliyun.bailian;
import com.alibaba.dashscope.audio.asr.transcription.*;
import com.alibaba.dashscope.exception.NoApiKeyException;
import com.alibaba.dashscope.utils.ApiKey;
import com.google.gson.*;
import java.io.BufferedReader;
import java.io.InputStreamReader;
import java.net.*;
import java.net.HttpURLConnection;
import java.util.Arrays;
import java.util.List;

public class TranscriptFilesByRestfulApi {

    public static void main(String[] args) throws NoApiKeyException {
        // create transcription params, use getDashScopeApiKey to get api key.
        TranscriptionParam param =
            TranscriptionParam.builder()
            .apiKey("your-dashscope-api-key")
            .model("paraformer-v1") // 'paraformer-8k-v1', 'paraformer-mtl-v1'
            .fileUrls(
                Arrays.asList(
                "https://dashscope.oss-cn-beijing.aliyuncs.com/samples/audio/sensevoice/rich_text_example_1.wav",
                "https://dashscope.oss-cn-beijing.aliyuncs.com/samples/audio/sensevoice/sample_video_poetry.mp4",
                "https://dashscope.oss-cn-beijing.aliyuncs.com/samples/audio/sensevoice/long_audio_demo_cn.mp3")
            .build();
        try {
            Transcription transcription = new Transcription();
            // post request to transcribe service
            TranscriptionResult result = transcription.asyncCall(param);
            // waiting for transcription finish
            result =
                    transcription.wait(
                            TranscriptionQueryParam.FromTranscriptionParam(param, result.getTaskId()));
            // get Transcription result after waiting
            List<TranscriptionTaskResult> taskResultList = result.getResults();
            if (taskResultList != null && !taskResultList.isEmpty()) {
                for (TranscriptionTaskResult taskResult : taskResultList) {
                    // get Transcription rersult url
                    String transcriptionUrl = taskResult.getTranscriptionUrl();
                    // get transcription result by http get
                    HttpURLConnection connection =
                            (HttpURLConnection) new URL(transcriptionUrl).openConnection();
                    connection.setRequestMethod("GET");
                    connection.connect();
                    BufferedReader reader =
                            new BufferedReader(new InputStreamReader(connection.getInputStream()));
                    // format json print
                    Gson gson = new GsonBuilder().setPrettyPrinting().create();
                    System.out.println(gson.toJson(gson.fromJson(reader, JsonObject.class)));
                }

            }
        } catch (Exception e) {
            System.out.println("error: " + e);
        }
        System.exit(0);
    }

    /**
     * Set your DashScope API key. More information: <a href="https://help.aliyun.com/document_detail/2712195.html">...</a>
     * In fact, if you have set DASHSCOPE_API_KEY in your environment variable,
     * you can ignore this, and the SDK will automatically get the api_key from the environment variable
     * */
    private static String getDashScopeApiKey() throws NoApiKeyException {
        String dashScopeApiKey = null;
        try {
            ApiKey apiKey = new ApiKey();
            dashScopeApiKey = apiKey.getApiKey(null); // Retrieve from environment variable.
        } catch (NoApiKeyException e) {
            System.out.println("No API key found in environment.");
        }
        if (dashScopeApiKey == null) {
            // If you cannot set api_key in your environment variable,
            // you can set it here by code
            dashScopeApiKey = "your-dashscope-api-key";
        }
        return dashScopeApiKey;
    }
}
